monkey/monkey_island/cc/ui/src/components/report-components/AttackReport.js

```diff
@@ -159,7 +159,9 @@ class AttackReport extends React.Component {
     }
     // modify techniques' messages
     for (const tech_id in techniques){
-      techniques[tech_id]['message'] = <div dangerouslySetInnerHTML={{__html: marked(techniques[tech_id]['message'])}} />;
+      if (typeof techniques[tech_id]['message'] === 'string') {
```

Comment:
Do we understand why the messages are objects instead of strings. What is the root cause of this issue? Are we just side-stepping it by checking the type?

Reply:
`marked()` modifies them in place. We need to change this behaviour(maybe there's a param for this? maybe we pass a copy of the string instead?)

Reply:
@VakarisZ Does `marked()` modify in place? All of the usage I can find implies that it returns a value, and javascript strings are immutable. **https://marked.js.org/**

Reply:
Oh, I think you mistyped and meant that `modifyTechniqueData()` modifies in-place. I think we're on the same page.